src/core/linux/SDL_dbus.c

```diff
@@ -287,6 +291,32 @@ SDL_DBus_CallVoidMethodInternal(DBusConnection *conn, const char *node, const ch
     return retval;
 }
 
+static SDL_bool
+SDL_DBus_CallWithBasicReply(DBusConnection *conn, DBusMessage *msg, const int expectedtype, void *result)
```

Comment:
(Pre-existing issue)

This is quite odd in D-Bus terms: every D-Bus method is meant to always return the same type, but this function accepts either `expectedtype` or a variant containing `expectedtype`.

For instance, if you call `org.freedesktop.DBus.Properties.Get` for a uint32 property, it should always return a variant containing an uint32 (and never a bare uint32), whereas if you call `org.freedesktop.Screensaver.Inhibit()` it should always return a bare uint32 (and never a variant containing a uint32).

However, there's no real harm done by SDL being more flexible than it needs to be...

Reply:
This function is not a 1:1 refactor of the existing code. Previously it was always expected to receive a variant containing `expectedtype`.
In order to support also methods that returns a bar basic type, I added this `if`.

Probably I should have been more clear in the commit message.

I need to think about how to refactor it without having to copy the entire function two times.

Reply:
Please don't consider this to be a merge blocker: it's entirely possible that the most pragmatic approach is to accept either one, even if more typical D-Bus clients wouldn't.

Reply:
I wasn't able to come up with a better design that didn't involve basically duplicating the entire function two times.

I left this as is for now.